### PR TITLE
Destination Kafka: Enforce SSL in cloud

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte-internal-issues/issues/846

as title: use the AdaptiveDestinationRunner to switch between the normal and strict-encrypt variants of KafkaDestination.

Currently the strict-encrypt version just requires SASL_SSL. After https://github.com/airbytehq/airbyte-internal-issues/issues/975 is closed, it should require either SSL or SSH tunneling.

The list of affected destinations is large b/c I fixed a typo in the adaptive runner. The first comment (https://github.com/airbytehq/airbyte/pull/17421#issuecomment-1262912237) correctly says the changes only affect destination-kafka.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-kafka/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 application {
-    mainClass = 'io.airbyte.integrations.destination.kafka.KafkaDestination'
+    mainClass = 'io.airbyte.integrations.destination.kafka.KafkaDestinationRunner'
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-kafka/build.gradle:8]: @tuliren is this the right usage? I.e.:
1. create a new StrictEncryptDestination class to wrap the original Destination class
2. create a new Runner wrapper class to call either the old or new Destinations classes
3. switch this mainClass to call the new Runner class

Comment (tuliren) [reply airbyte-integrations/connectors/destination-kafka/build.gradle:8]: Yes! Nice.

Comment (tuliren) [reply airbyte-integrations/connectors/destination-kafka/build.gradle:8]: You can test it by:
1. Launch a normal local instance, and the non-strict encrypt spec should show up.
2. Add `DEPLOYMENT_MODE=CLOUD` in the `.env` file and launch the local instance again, only the strict encrypt spec should show up.

Based on my experience, one tricky thing is that only the `docker-compose` command will respect the `.env` file. The newer `docker compose` will not.

Comment (edgao) [reply airbyte-integrations/connectors/destination-kafka/build.gradle:8]: fyi you can also just pass the env var on the command line, e.g. `DEPLOYMENT_MODE=CLOUD VERSION=dev docker-compose up`. In my testing it also works with the newer `docker compose` :)

Comment (tuliren) [reply airbyte-integrations/connectors/destination-kafka/build.gradle:8]: Cool. That's good to know.

**Pull request closed without merging**